Plugins/NativeCamera/Source/Android/CameraWrappers.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <android/api-level.h>
+#include <dlfcn.h>
+#include <map>
+
+// The camera API's in the NDK are only supported in API24 and higher. We
+// use dlopen and dlsym to link the camera lib at runtime only if the current
+// device supports it. In order to avoid redefining the type information the
+// below macro's forcefully import the camera header files as if the min supported
+// SDK was set to 24.
+#define __ORIG_ANDROID_API__ __ANDROID_API__
+#undef __ANDROID_API__
+#define __ANDROID_API__ 24
+namespace API24 {
+#include <camera/NdkCameraManager.h>
+    #include <camera/NdkCameraCaptureSession.h>
+    #include <camera/NdkCameraDevice.h>
+    #include <camera/NdkCameraError.h>
+    #include <camera/NdkCameraManager.h>
+    #include <camera/NdkCameraMetadata.h>
+    #include <camera/NdkCameraMetadataTags.h>
+    #include <camera/NdkCameraWindowType.h>
+    #include <camera/NdkCaptureRequest.h>
+}
+#undef __ANDROID_API__
+#define __ANDROID_API__ __ORIG_ANDROID_API__
+#undef __ORIG_ANDROID_API__
+
+// Helper macro to make calling camera NDK api's more type safe
+#define GET_CAMERA_FUNCTION(function) reinterpret_cast<decltype(&API24::function)>(Babylon::Plugins::GetCameraDynamicFunction(#function))
+
+namespace
+{
+    static const int API_LEVEL{ android_get_device_api_level() };
+
+    // Load the NDK camera lib dynamically. When running on OS 6.0 and below this will return nullptr,
+    // on OS 7.0 and up this should return a pointer to access the library using dlsym. It is technically fine
+    // to call dlopen when the API_LEVEL is below 24, but it's wasted effort on those devices.
+    static void* libCamera2NDK{ API_LEVEL >= 24 ? dlopen("libcamera2ndk.so", RTLD_NOW): nullptr };
+
+    static std::map<std::string, void*> CameraDynamicFunctions{};
+}
+
+namespace Babylon::Plugins {
+    static inline void* GetCameraDynamicFunction(const char* functionName) {
+        if (!libCamera2NDK)
+        {
+            return nullptr;
+        }
+        auto cameraDynamicFunctionIt{CameraDynamicFunctions.find(functionName)};
+        if (cameraDynamicFunctionIt == CameraDynamicFunctions.end())
+        {
+            cameraDynamicFunctionIt = CameraDynamicFunctions.emplace(functionName, dlsym(libCamera2NDK, functionName)).first;
+        }
```

Comment:
Do we know for sure if this is faster than just calling `dlsym` every time? Is this function slow?

Reply:
No we don't actually know. If we just store the individual function pointers (like you suggest below), then we'd only call dlsym once per function anyway. We could maybe test that out, or save it for a future change. 🤷🏼‍♂️

Reply:
I didn't test it out so take this with a grain of salt. The [docs for dlsym](https://man7.org/linux/man-pages/man3/dlsym.3.html) call out that it will do a breadth first search for the symbol first through the library passed in and then through any other shared/global libraries loaded with `dlopen`. It doesn't indicate any caching is in place, so I would say it's a good call to do some form of caching on our own side.

Reply:
Either way, it would be interesting to know, but it's totally fine to merge without addressing this.